pkgs/servers/web-apps/hedgedoc/default.nix

```
@@ -1,73 +1,94 @@
 { lib
 , stdenv
-, fetchzip
-, makeWrapper
-, which
+, fetchFromGitHub
+, gitMinimal
+, cacert
+, yarn
+, makeBinaryWrapper
 , nodejs
-, mkYarnPackage
-, fetchYarnDeps
+, nodePackages
 , python3
 , nixosTests
-}:
+}: let
 
-mkYarnPackage rec {
-  pname = "hedgedoc";
-  version = "1.9.7";
-
-  # we use the upstream compiled js files because yarn2nix cannot handle different versions of dependencies
-  # in development and production and the web assets muts be compiled with js-yaml 3 while development
-  # uses js-yaml 4 which breaks the text editor
-  src = fetchzip {
-    url = "https://github.com/hedgedoc/hedgedoc/releases/download/${version}/hedgedoc-${version}.tar.gz";
-    hash = "sha256-tPkhnnKDS5TICsW66YCOy7xWFj5usLyDMbYMYQ3Euoc=";
+  version = "1.9.8";
+
+  src = fetchFromGitHub {
+    owner = "hedgedoc";
+    repo = "hedgedoc";
+    rev = version;
+    hash = "sha256-gp1TeYHwH7ffaSMifdURb2p+U8u6Xs4JU4b4qACEIDw=";
   };
 
-  nativeBuildInputs = [ which makeWrapper ];
-  extraBuildInputs = [ python3 ];
+  offlineCache = stdenv.mkDerivation {
+    name = "hedgedoc-${version}-offline-cache";
+    inherit src;
+
+    nativeBuildInputs = [
+      cacert # needed for git
+      gitMinimal # needed for a dep
+      nodePackages.npm # needed for a dep
+      yarn
+    ];
+
+    HOME = "$TMPDIR";
+    buildPhase = ''
+      yarn config set enableTelemetry 0
+      yarn config set cacheFolder $out
+      yarn
+    '';
+
+    outputHashMode = "recursive";
+    outputHash = "sha256-4TPc9J9u0rO9WfZVbFWwVAM5WY3n5zJdhIlNYiZNRoM=";
+  };
 
-  packageJSON = ./package.json;
-  yarnFlags = [ "--production" ];
+in stdenv.mkDerivation rec {
```

How about using a better overriding practice like the `finalAttrs` pattern?

```suggestion
in stdenv.mkDerivation (finalAttrs: {
```

---

I don't know if I will have time in the next couple of days, if anyone is interested, they can just push the branch so this gets through faster

---

Simply remove the rec and then it's fine.